tests/granulate_utils/test_futures.py

```diff
@@ -0,0 +1,36 @@
+import time
+
+import pytest
+
+from granulate_utils.futures import call_in_parallel
+
+
+def wait(seconds: int) -> int:
+    time.sleep(seconds)
+    return seconds
+
+
+def test_call_in_parallel_sanity() -> None:
+    assert 1 == next(call_in_parallel((lambda: wait(1), lambda: wait(2), lambda: wait(3)), timeout=5)).result()
+    assert 1 == next(call_in_parallel((lambda: wait(3), lambda: wait(2), lambda: wait(1)), timeout=5)).result()
+    for i, future in enumerate(call_in_parallel((lambda: wait(0), lambda: wait(1), lambda: wait(2)), timeout=5)):
+        assert i == future.result()
+
+
+def test_call_in_parallel_timeout() -> None:
+    with pytest.raises(TimeoutError, match="futures unfinished"):
+        next(call_in_parallel((lambda: wait(2), lambda: wait(2)), timeout=1)).result()
+
+
+def test_call_in_parallel_exception_handling() -> None:
```

Comment:
One more test I can think of (and actually wonder what the result would be) is if we have one that throws and then we call next again.
This happens in your change and actually I wonder if it's correct, you built it as 
```
for x in call_in_parallel:
   try:
     x.result()
   except ....
```

To my understanding the `next` is called in the `for` line, so outside of the `try`. It's the `result` call that's raising I suppose, but let's test it